### ISSUE TITLE
Invalid vector<T> subscript error

### ISSUE THREAD

**Issue by Speigy** [moneymanagerex/moneymanagerex#4129]
![Screenshot 2022-01-27 105059](https://user-images.githubusercontent.com/39367770/151344943-8612fc97-fb2e-48b9-a13a-9aba5bf9b7c4.jpg)
According to my previous rquest for an answer, my request has been marked as closed???  I like this program but it has significnt flaws. First, it doesn't do a year end roll over, a standard concept in accounting procedures. Secondly, support is hit and miss, you have to wait for the 'geeks' to pick up on the error. I have not donated yet; I will not until these issues are sorted. I appreciate the background of the software, and that it is free. Personally I would be happy to pay if there was a dedicated support team.
![Screenshot 2022-01-27 105059](https://user-images.githubusercontent.com/39367770/151345688-0841466f-c5a1-4315-92d0-0fca3c94aaa3.jpg)

**Comment by whalley** [comment]
I closed it as you never responded to the request to supply steps to re-produce, and I did state that you should re-open when you could supply this. Just creating another issue, again not mentioning how to reproduce the error is not helpful as we have no information on which to investigate as the error could be being generated anywhere in the code.

As requested before: **Please describe the steps to re-produce the error condition.**

**Comment by tactilis** [comment]
@Speigy As you have noted, MMEX is free, open source software.

It is developed and supported by people who volunteer their time. They are not obliged to develop new features, investigate and fix bugs, or provide any support at all... but they do so, to the best of their ability in the limited time they have available.

You say:
> Personally I would be happy to pay if there was a dedicated support team.

I'm afraid that you just haven't thought what it would cost to have a dedicated support team. How much would you be happy to pay?

Obviously, you are not obliged to use MMEX, and are at liberty to move over to a paid-for commercial product if that better suits your needs.

The main developers of MMEX have not yet been able to reproduce the error that you have reported. If they cannot reproduce it, then they are unable find what is wrong in the code and fix it for you. In the original issue report and in this issue @whalley asked you to provide information on the steps that cause the error to occur as this may help us home in on the problem.

We all realise that problems are very frustrating for users but we rely on the users observing the exact conditions that lead up to a fault and reporting them as clearly as possible, especially if the problem only affects one or just a few users. It is not necessary to be a "geek", as you put it, to do this.

We are happy to help hand-hold you through this but people will be put off doing so if you write in the manner that you did above.

**Comment by whalley** [comment]
> I have read all the background of what MMEX is all about and how it has come about. I am being asked to reproduce the problem. I can't! It appears in every month forward. I insert transaction still year end. Every month has this error. Perhaps someone could tell me how to create a new databse and import catergories, recurring transactions, payees etc. I can't find a way of doing it. Someone suggested if I don't like it buy a commercial package. Not a very satifactory response.

Can you please describe what exactly you are doing prior to the error occurring.

> It appears in every month forward.
> Every month has this error. 

You get the error once a month?

> I insert transaction still year end.

No idea what this means.

**Comment by vomikan** [comment]
It seems some asset account has been created for some asset. Then it has been removed but transfer transaction to the asset account remained. 
This is a rather complicated problem. No one understands the concept of linking accounts, such as investment and asset accounts.
In this case, I recommend deleting the defective transaction. Instead, create a transaction to update the balance.

**Comment by vomikan** [comment]
https://github.com/moneymanagerex/moneymanagerex/commit/61db7fa573d2e6acb8a094fea77b321b5ba54fbd

**Comment by whalley** [comment]
> It seems some asset account has been created for some asset. Then it has been removed but transfer transaction to the asset account remained.
> This is a rather complicated problem. No one understands the concept of linking accounts, such as investment and asset accounts.
> In this case, I recommend deleting the defective transaction. Instead, create a transaction to update the balance.

@vomikan - Is this in response https://github.com/moneymanagerex/moneymanagerex/issues/3727 raised by @TonySLondon - do you have more info on this case or are you assuming it is the same condition?

**Comment by vomikan** [comment]
I'm also very confused about these cases. You can get a invalid vector in any part of the code.
In case 3727 it was ```translink_list.at(0)```

**Comment by tactilis** [comment]
> Is this in response #3727 raised by @TonySLondon - do you have more info on this case or are you assuming it is the same condition?

I may be wrong but I think @TonySLondon is the same person as @Speigy.  If so, they are reports about the same MMEX problem. 

@Speigy Could you confirm whether this is the case please?

**Comment by whalley** [comment]
> I may be wrong but I think @TonySLondon is the same person as @Speigy. If so, they are reports about the same MMEX problem.

Having reviewed the screenshots you may indeed be right !!!

**Comment by whalley** [comment]
@Speigy Is this issue now resolved in v1.5.12?

**Comment by whalley** [comment]
@Speigy Due to lack of feedback, I'm going to assume this issue is now resolved.

**Comment by wolfgang-ch** [comment]
When I select a preset, then this error occurs and when I select the "Load this Preset..." then I cannot open the import dialog any more.


![image](https://github.com/moneymanagerex/moneymanagerex/assets/1283445/3af877bd-3cc7-4b83-aadd-06ea73d46764)

Money Manager Ex
• Version: 1.6.4 64-bit
• Built on: Jun 20 2023 18:27:14
• Database version: 18 (aes128cbc)
• Git commit: 6b4cc69 (2023-06-20)

MMEX is using the following support products:
• wxWidgets 3.2.1 (wxMSW 10.0)
• wxSQLite3 4.9.1 (SQLite 3.40.0)
• RapidJSON 1.1.0
• Lua 5.3.5
• libcurl/7.82.0-DEV Schannel
• gettext 0.19.8.1
• apexcharts.js

Built with:
• CMake 3.16.2
• MSBuild 15.9.21.664
• Microsoft Visual Studio 15.0
• Microsoft Visual C++ 19.16.27035.0
• Windows SDK 10.0.14393.0

Running on:
• Windows 10 (build 19045), 64-bit edition
• English (Switzerland) (windows-1252)
• 3840x2160 32bit 96x96ppi

**Comment by whalley** [comment]
See: https://github.com/moneymanagerex/moneymanagerex/issues/6373#issuecomment-1865016629
